bot/cogs/error_handler.py

```diff
@@ -0,0 +1,89 @@
+import logging
+
+from discord.ext.commands import (
+    BadArgument,
+    BotMissingPermissions,
+    CommandError,
+    CommandInvokeError,
+    CommandNotFound,
+    NoPrivateMessage,
+    UserInputError,
+)
+from discord.ext.commands import Bot, Context
+
+from bot.api import ResponseCodeError
+
+
+log = logging.getLogger(__name__)
+
+
+class ErrorHandler:
+    """Handles errors emttted from commands."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    async def on_command_error(self, ctx: Context, e: CommandError):
+        command = ctx.command
+        parent = None
+
+        if command is not None:
+            parent = command.parent
+
+        if parent and command:
+            help_command = (self.bot.get_command("help"), parent.name, command.name)
+        elif command:
+            help_command = (self.bot.get_command("help"), command.name)
+        else:
+            help_command = (self.bot.get_command("help"),)
+
+        if hasattr(command, "on_error"):
+            log.debug(f"Command {command} has a local error handler, ignoring.")
+            return
+
+        if isinstance(e, CommandNotFound) and not hasattr(ctx, "invoked_from_error_handler"):
+            tags_get_command = self.bot.get_command("tags get")
+            ctx.invoked_from_error_handler = True
+
+            # Return to not raise the exception
+            return await ctx.invoke(tags_get_command, tag_name=ctx.invoked_with)
+        elif isinstance(e, BadArgument):
+            await ctx.send(f"Bad argument: {e}\n")
+            await ctx.invoke(*help_command)
+        elif isinstance(e, UserInputError):
+            await ctx.invoke(*help_command)
```

Comment:
Should a message still be sent here?

Reply:
Yes, I think it makes sense - if the user did something wrong, show help for the command. Or does that not make sense?

Reply:
I mean a message in addition to the help command.

Reply:
Added in 74f4162d2859333909c74b038ee3a1dcaee65c9c.

Reply:
In what kind of situations is this exception raised? When an argument can't be parsed, like an invalid value for a duration, BadArgument gets raised and that one does already provide feedback about which argument is problematic.

I would only add a specific message to the user if it actually conveys useful information, but I'm not sure when this exception even happens.